lib/scala/semver/src/main/java/org/enso/semver/SemVer.java

```diff
@@ -0,0 +1,149 @@
+package org.enso.semver;
+
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import scala.util.Failure;
+import scala.util.Success;
+import scala.util.Try;
+
+/**
+ * A facade hiding the implementation of the Semantic Versioning used in Enso.
+ */
+public class SemVer {
+
+    private long major;
+    private long minor;
+    private long patch;
+    private Optional<String> preRelease;
+    private Optional<String> prefix;
+
+    private static final Pattern regex = Pattern.compile("([a-z]+-)?(\\d|[1-9]\\d+)\\.(\\d|[1-9]\\d+)\\.(\\d|[1-9]\\d+)(-[a-z]+)?");
+
+    private SemVer(long major, long minor, long patch, Optional<String> preRelease, Optional<String> prefix) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+        this.preRelease = preRelease;
+        this.prefix = prefix;
+    }
+
+    private SemVer(long major, long minor, long patch, Optional<String> preRelease) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+        this.preRelease = preRelease;
+        this.prefix = Optional.empty();
+    }
+
+    private SemVer(long major, long minor, long patch) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+        this.preRelease = Optional.empty();
+        this.prefix = Optional.empty();
+    }
+
+    public static Try<SemVer> parse(String version) {
```

Comment:
`Try`!? It is good this class is written in Java, but it doesn't feel like real Java when the Scala concepts remain. Real Java API would return `SemVer` and `throw` an exception.

Reply:
And we would have to catch that exception in every Scala code that calls it. :shrug: Till most of the Scala code calls this method I'd prefer to stick to `Try`.